src/python/gudhi/datasets/remote.py

```diff
@@ -204,20 +241,85 @@ def fetch_bunny(file_path = None, accept_license = False):
+_activities_license_checksum = "f5ce6749fa9d5359d7b0c4c37d0b61e5d9520f9494cd53be94295d3967ee4023"
+
+
+def fetch_daily_activities(file_path=None, subset=None, accept_license=False):
+    """
+    Load a subset of the Daily and Sports Activities dataset. This dataset comes from
+    https://archive.ics.uci.edu/ml/datasets/daily+and+sports+activities (CC BY 4.0 license).
+
+    Note that if the dataset already exists in the target location, it is not downloaded again,
+    and the corresponding dataset is read from cache.
+
+    Parameters
+    ----------
+    file_path : string
+        Full path of the downloaded file including filename.
+
+        Default is None, meaning that it's set to "data_home/points/activities/activities_p1_left_leg.npy".
+        In this case, the LICENSE file would be downloaded as "data_home/points/activities/activities.LICENSE".
+
+        The "data_home" directory is set by default to "~/gudhi_data",
+        unless the 'GUDHI_DATA' environment variable is set.
+
+    subset : string
+        This argument allows to download the following subsets:
+         * 'cross_training' Only left leg magnetometer of cross training activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'jumping' Only left leg magnetometer of jumping activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'stepper' Only left leg magnetometer of stepper activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * 'walking' Only left leg magnetometer of walking activity performed by the person 1. It contains 7.500 vertices in dimension 3.
+         * None (default value) This dataset contains 30.000 vertices in dimension 3 + activity type column (`14.` for 'cross_training', `18.` for 'jumping', `13.` for 'stepper', or `9.` for 'walking')
+
+    accept_license : boolean
+        Flag to specify if user accepts the file LICENSE and prevents from printing the corresponding license terms.
+
+        Default is False.
+
+    Returns
+    -------
+    points: numpy array
+        Depending on subset value:
+         * Array of shape (7.500, 3, dtype = float).
+
+        Or
+         * Array of shape (30.000, 4, dtype = float) when `subset is None`. 
+    """
+    file_url = (
+        "https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/activities/activities_p1_left_leg.npy"
+    )
+    file_checksum = "ff813f717dbd3c8f8a95e59a7d8496d0b43c440e85a4db1f2b338cbfa9c02f25"
+    gudhi_data_set_path = "points/activities/activities_p1_left_leg.npy"
+
+    archive_path = _get_archive_path(file_path, gudhi_data_set_path)
+    if not exists(archive_path):
+        _fetch_remote(file_url, archive_path, file_checksum)
+        license_path = join(split(archive_path)[0], "activities.LICENSE")
+        _fetch_remote_license(_activities_license_url, license_path, _activities_license_checksum, accept_license)
+
+    ds = np.load(archive_path)
```

Comment:
Ideally, once loaded the first time we would store this somewhere, so that the next call can reuse the same ds. For instance split the production of this ds into a function marked with @functools.cache. But that's not important, let's ignore that.

Reply:
I did this change [da87405](https://github.com/GUDHI/gudhi-devel/pull/690/commits/da87405baacbc8e6aa95ebd86e786ba93678b56d) to make unitary tests faster.
Strangely, adding @functools.cache does not make it faster... :thinking:

Reply:
Ok, I got it ! I cached the numpy load result on [26cdd10](https://github.com/GUDHI/gudhi-devel/pull/690/commits/26cdd10b9a97bcbf3d56db1455cd3c89c7c99214) and after testing it again with strace, I can see the file is no more loaded at each fetch call.

I also improved subset condition test on [958837f](https://github.com/GUDHI/gudhi-devel/pull/690/commits/958837f71e0a51a4e625c64ea43cdd52b8a7150f)